jaeger/charts/linkerd-jaeger/README.md

```diff
@@ -86,6 +87,14 @@ Kubernetes: `>=1.16.0-0`
 | linkerdVersion | string | `"linkerdVersionValue"` |  |
 | namespace | string | `"linkerd-jaeger"` |  |
 | nodeSelector | object | `{"beta.kubernetes.io/os":"linux"}` | Default nodeSelector section, See the [K8S documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector) for more information |
+| opentelemetry.collector.config | object | `{"exporters":{"jaeger":{"endpoint":"jaeger.linkerd-jaeger:14250","insecure":true}},"extensions":{"health_check":null},"processors":{"batch":null},"receivers":{"jaeger":{"protocols":{"grpc":null,"thrift_binary":null,"thrift_compact":null,"thrift_http":null}},"opencensus":null,"otlp":{"protocols":{"grpc":null,"http":null}},"zipkin":null},"service":{"extensions":["health_check"],"pipelines":{"traces":{"exporters":["jaeger"],"processors":["batch"],"receivers":["otlp","opencensus","zipkin","jaeger"]}}}}` | OpenTelemetry Collector config, See the [Configuration docs](https://opentelemetry.io/docs/collector/configuration/) for more information |
+| opentelemetry.collector.image.name | string | `"otel/opentelemetry-collector"` |  |
+| opentelemetry.collector.image.pullPolicy | string | `"Always"` |  |
+| opentelemetry.collector.image.version | string | `"0.27.0"` |  |
+| opentelemetry.collector.jaegerAddr | string | `"jaeger.linkerd-jaeger:14250"` | address of the jaeger backend to send traces to |
+| opentelemetry.collector.nodeSelector | object | `{"beta.kubernetes.io/os":"linux"}` | NodeSelector section, See the [K8S documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector) for more information |
+| opentelemetry.collector.tolerations | string | `nil` | Tolerations section, See the [K8S documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) for more information |
+| opentelemetry.enabled | bool | `false` | If enabled then OpenTelemetry collector is used instead of OpenCensus Collector, See the [OpenTelemetry Collector documentation](https://opentelemetry.io/docs/collector/) for more information |
```

Comment:
I'd like to think through the path to making this on by default... if we're going to add configuration to the project, we need a path to removing the older options so we're not stuck supporting them forever.

From my point of view, the choice of collector in this extension is mostly an implementation detail -- i.e. we should be able to swap out the OC collector for the OT collector without any user-facing choices. Furthermore, your change looks purely additive from the Helm perspective, so it's not clear to me how this really introduces any backwards incompatibility. Is there really a problem with enabling this by default?

Reply:
Try to add more information:

1. No, it is not just additive because collector deployment and configuration will be replaced when the toggle on.
2. The next issue is configuration because collectors are not compatible this way. But as I see, OC configuration is not modifiable (only jaeger address can be modified), so it is not a problem if nobody overwrites configuration some way.
3. OpenTelemetry Jaeger export does not support thrift protocol (only the contrib version does) that is used in OC collector configuration. If anyone uses this protocol with their own Jaeger installation, it could break traces delivery (and it can be detected only by logs in the collector).

If this is not a problem, I can rework and make it default or even eliminate the OC collector.

Reply:
I agree with @olix0r that the collector is an implementation detail and would prefer we just replace the OC collector entirely.  The jaeger extension is supposed to be more of a reference architecture than something that is intended to work for all users anyway.

Reply:
Ok, with this commit (https://github.com/linkerd/linkerd2/commit/b842652b1093d0c87fc690e94f6ec1c793908173) users can choose their own configuration, so I rework my PR and now it is just replaces collector.